src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -1,112 +1,58 @@
-imports:
-  - { resource: 'form_type/*' }
-
 services:
   _defaults:
-    public: true
+    public: false
+    autowire: true
+    autoconfigure: true
+
+  _instanceof:
+    # Add the console.command tag to all services defined in this file having this type
+    PrestaShopBundle\Form\Admin\Type\TranslatorAwareType:
+      bind:
+        $locales: "@=service('prestashop.adapter.legacy.context').getLanguages()"
 
-  form.type.date_picker:
-    class: PrestaShopBundle\Form\Admin\Type\DatePickerType
+  PrestaShopBundle\Form\Admin\Type\DatePickerType:
     arguments:
       - '@prestashop.bundle.form.data_transformer.arabic_to_latin_digit'
-    tags:
-      - { name: form.type }
 
-  form.type.navigation_tab:
-    class: PrestaShopBundle\Form\Admin\Type\NavigationTabType
+  PrestaShopBundle\Form\Admin\Type\NavigationTabType:
     arguments:
       - '@logger'
       - '@=service("prestashop.adapter.environment").isDebug()'
-    tags:
-      - { name: form.type }
 
-  form.type.email:
-    class: PrestaShopBundle\Form\Admin\Type\EmailType
+  PrestaShopBundle\Form\Admin\Type\EmailType:
     arguments:
       - '@prestashop.bundle.form.data_transformer.idn_converter'
-    tags:
-      - { name: form.type }
 
-  form.type.date_range:
-    class: PrestaShopBundle\Form\Admin\Type\DateRangeType
+  PrestaShopBundle\Form\Admin\Type\DateRangeType:
     arguments:
       - "@translator"
       - "@form.form_cloner"
-    tags:
-      - { name: form.type }
 
-  form.type.product.simple_category:
-    class: PrestaShopBundle\Form\Admin\Category\SimpleCategory
+  PrestaShopBundle\Form\Admin\Category\SimpleCategory:
     arguments:
       - "@translator"
       - "@prestashop.adapter.data_provider.category"
-    tags:
-      - { name: form.type }
 
-  form.type.product.categories_tree:
-    class: 'PrestaShopBundle\Form\Admin\Type\ChoiceCategoriesTreeType'
-    parent: 'form.type.common_type'
-    public: true
-    tags:
-      - { name: form.type }
+  PrestaShopBundle\Form\Admin\Type\ChoiceCategoriesTreeType:
 
-  form.type.product.translate:
-    class: 'PrestaShopBundle\Form\Admin\Type\TranslateType'
-    parent: 'form.type.common_type'
+  PrestaShopBundle\Form\Admin\Type\TranslateType:
     arguments:
-      - '@router.default'
-      - "@=service('prestashop.adapter.legacy.configuration').getBoolean('PS_BO_ALLOW_EMPLOYEE_FORM_LANG')"
-      - "@=service('prestashop.adapter.legacy.context').getContext().cookie.employee_form_lang"
-      - "@=service('prestashop.adapter.legacy.configuration').getInt('PS_LANG_DEFAULT')"
-    public: true
-    tags:
-      - { name: form.type }
+      $saveFormLocaleChoice: "@=service('prestashop.adapter.legacy.configuration').getBoolean('PS_BO_ALLOW_EMPLOYEE_FORM_LANG')"
+      $defaultFormLanguageId: "@=service('prestashop.adapter.legacy.context').getContext().cookie.employee_form_lang"
+      $defaultShopLanguageId: "@=service('prestashop.adapter.legacy.configuration').getInt('PS_LANG_DEFAULT')"
```

Comment:
Should we use generic binding instead, to mutualize the definition:
```yaml
services:
  _defaults:
    public: false
    autowire: true
    autoconfigure: true
    bind:
        $saveFormLocaleChoice: "@=service('prestashop.adapter.legacy.configuration').getBoolean('PS_BO_ALLOW_EMPLOYEE_FORM_LANG')"
        ...
```

What do you think?

Reply:
That makes sense, but for the existing services definitions, the same attribute can have many names, sometimes we can have `currency`, `currencyIsoCode` or `currencyIsoCode` for the same parameter inside a constructor.

I agree that this is weird and this should be avoided, but we can have some side effects  renaming them. And that would make this PR way too complicated to review and maintain. This can be done in a second time.

Reply:
Ok for me. Plz consider the following message from @mflasquin and I'll approve 👍